File: medium/2966. Divide Array Into Arrays With Max Difference/2966. Divide Array Into Arrays With Max Difference.py
from typing import List


class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        n = len(nums)
        nums.sort()

        res: List[List[int]] = []

        for i in range(0, n - 2, 2):
            candidate = nums[i : i + 3]

            if len(candidate) < 3 or abs(candidate[0] - candidate[2]) > k:
                return []

            res.append(candidate)

        return res

File: medium/2966. Divide Array Into Arrays With Max Difference/2966. Divide Array Into Arrays With Max Difference.ts
function divideArray(nums: number[], k: number): number[][] {
    nums.sort((a, b) => a - b)

    const n = nums.length
    const res: number[][] = []

    for (let i = 0; i <= n - 2; i += 3) {
        if (nums[i + 2] - nums[i] > k) return []
        res.push(nums.slice(i, i + 3))
    }

    return res
};